pages/admin/lessons/[lessonId]/[pageName]/index.tsx

```diff
@@ -0,0 +1,179 @@
+import { gql, useQuery } from '@apollo/client'
+import { GetAppProps, withGetApp } from '../../../../../graphql'
+import { get, sortBy, toNumber, toUpper } from 'lodash'
+import React, { useMemo, useState } from 'react'
+import AdminLessonNav from '../../../../../components/admin/lessons/AdminLessonSideNavLayout'
+import AdminLessonSideNav from '../../../../../components/admin/lessons/AdminLessonSideNav'
+import AdminModuleInputs, {
+  Props
+} from '../../../../../components/admin/lessons/AdminModuleInputs'
+import Breadcrumbs from '../../../../../components/Breadcrumbs'
+import styles from '../../../../../scss/modules.module.scss'
+import navStyles from '../../../../../scss/adminLessonNav.module.scss'
+import { AdminLayout } from '../../../../../components/admin/AdminLayout'
+import { useRouter } from 'next/router'
+import Link from 'next/link'
+
+const MAIN_PATH = '/admin/lessons'
+
+const MODULES = gql`
+  query {
+    modules {
+      id
+      name
+      content
+      lesson {
+        id
+      }
+      order
+    }
+  }
+`
+
+type Module = {
+  id: number
+  name: string
+  content: string
+  lesson: { id?: number }
+  order: number
+}
+type Modules = Module[]
+
+type ReqParam = string | string[] | undefined
+
+type ContentProps = {
+  pageName: ReqParam
+  modules: Modules
+  lessonId: number
+  refetch: Props['refetch']
+}
+
+const extractLessonOrder = (x: string) => x.split('').slice(2).join()
+
+const Content = ({ pageName, modules, lessonId, refetch }: ContentProps) => {
+  const [selectedIndex, setSelectedIndex] = useState(-1)
+  const onAddItem = () => setSelectedIndex(-1)
+  const onSelect = (item: Omit<Module, 'order'>) => setSelectedIndex(item.id)
+
+  if (pageName !== 'modules') {
+    return (
+      <h1>
+        For now, you can only access <code>/modules</code> page
+      </h1>
+    )
+  }
+
+  return (
+    <div className={styles.container__modulesPanel}>
+      <AdminLessonSideNav
+        title="Modules"
+        items={modules}
+        onAddItem={onAddItem}
+        onSelect={onSelect}
+        selectedIndex={selectedIndex}
+      />
+      <div className={styles.container__modulesPanel__inputs}>
+        <AdminModuleInputs
+          lessonId={lessonId}
+          refetch={refetch}
+          module={modules.find(module => module.id === selectedIndex)}
+        />
+      </div>
+    </div>
+  )
+}
+
+const Lessons = ({ data }: GetAppProps) => {
+  const router = useRouter()
+  const { pageName, lessonId } = router.query
+
+  const { lessons } = data
+  const { data: modulesData, refetch } = useQuery<{ modules: Modules }>(MODULES)
+
+  const lesson = useMemo(() => {
+    if (lessons) {
+      const lessonFromParam = lessons.find(
+        e => toNumber(extractLessonOrder(e.slug)) === toNumber(lessonId)
```

Comment:
shouldn't we be comparing the slugs of each lesson? `e.slug === lessonSlug`

Reply:
But `lessonSlug` is a number that comes from the parameter.